app/org/sagebionetworks/bridge/services/StudyService.java

```diff
@@ -68,25 +74,33 @@
 import org.sagebionetworks.bridge.models.studies.StudyIdentifier;
 import org.sagebionetworks.bridge.models.upload.UploadFieldDefinition;
 import org.sagebionetworks.bridge.models.upload.UploadValidationStrictness;
+import org.sagebionetworks.bridge.services.email.BasicEmailProvider;
 import org.sagebionetworks.bridge.validators.StudyParticipantValidator;
 import org.sagebionetworks.bridge.validators.StudyValidator;
 import org.sagebionetworks.bridge.validators.Validate;
 
 @Component("studyService")
 public class StudyService {
-
     private static Logger LOG = LoggerFactory.getLogger(StudyService.class);
 
+    private static final String BASE_URL = BridgeConfigFactory.getConfig().get("webservices.url");
+    static final String CONFIG_KEY_SUPPORT_EMAIL_PLAIN = "support.email.plain";
+    private static final String CONSENT_NOTIFICATION_VERIFY_EMAIL_URL =
+            "%s/mobile/verifyConsentNotificationEmail.html?study=%s&sptoken=%s";
```

Comment:
I just put some effort into moving all these workflow emails into AccountWorkflowService, so we could remove sending mail directly from this service. It allows us to reuse things like BASE_URL, formatting and cache services in that one service. Can this logic be moved there?

Reply:
Maybe we should rename that the WorkflowService or something, If we'd no longer just using it for account-specific emails.

Reply:
As per IM conversation, from a code consolidation perspective, the only thing we'd be able to re-use is BASE_URL and createTimeLimitedToken(). Maybe saveVerification() and restoreVerification() can be refactored to take in class and expiration time to make it generalized.

From a code organization purpose, I can see a benefit here if we renamed AccountWorkflowService to something like ChannelVerificationService. Regardless, as per IM conversation, we decided this was best left for a future refactor.